### PR TITLE
Add `onInstall` and `onUninstall` hooks

### PR DESCRIPTION
**Pull request by grant0417**
Add hooks for `onInstall` and `onUninstall` of a plugin

### CODE REVIEW

**Review by sullivan-sean — APPROVED** [review]

**Comment by mschrage** [comment]
This is such a good addition @grant0417!

File: index.d.ts [hunk 1]
```diff
@@ -33,7 +33,37 @@ declare namespace Fig {
...
+      /** Check that commands exists */
+      commandExists?: InstallationScript<string[]>;
+      /** Check that files/folders exists */
+      fileExists?: InstallationScript<string[]>;
+      /** Check a commands output */
+      commandOutput?: {
+        /** The command to run to check the output */
+        command: InstallationScript<string>;
+        /** The expected output */
+        output: InstallationScript<string>;
```
Comment (mschrage) [inline index.d.ts:50]: Not sure I understand the API here. Are these scripts that return a boolean value? Or are they returning a list of files, executables that is validated by the CLI? (I assume the later?)

**Review by mschrage — APPROVED** [review]
This is such a smart addition. Now plugins can install executables for themselves!

File: index.d.ts (hunk 1 above)
Comment (grant0417) [reply index.d.ts:50]: The later

**Pull request merged**